modules/disk-uploader/pkg/vmexport/vmexport_test.go

```diff
@@ -186,4 +191,68 @@ var _ = Describe("VMExport", func() {
 			Expect(err).To(MatchError("no links found in VirtualMachineExport status"))
 		})
 	})
+
+	Describe("GetLabelsFromExportSource", func() {
+		It("should return labels from datavolume", func() {
+			_, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(namespace).Create(context.Background(), &cdiv1beta1.DataVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example-volume",
+					Namespace: namespace,
+					Labels: map[string]string{
+						"instancetype.kubevirt.io/default-instancetype": "cx1.2xlarge",
+						"instancetype.kubevirt.io/default-preference":   "fedora",
+					},
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			labels, err := vmexport.GetLabelsFromExportSource(virtClient, "VirtualMachine", namespace, name, "example-volume")
```

Comment:
Make it a table for "VirtualMachine" and "VirtualMachineSnapshot"? Add test for all cases where no DV/PVC could be found?

Reply:
Done.